tests/data/EventFactory.ts

```diff
@@ -48,7 +72,41 @@ export class EventFactory {
     const duration = FactoryUtils.getRandomNumber(30, 150, 30);
     const start = moment().subtract(days, 'days').hour(hour);
     const end = moment(start.valueOf()).add(duration, 'minutes');
-    return [new Date(start.valueOf()), new Date(end.valueOf())];
+    return [start.toDate(), end.toDate()];
+  }
+
+  private static randomPastTime(daysAgo: number): [Date, Date] {
```

Comment:
Can we reuse code between these `randomTime` methods? Maybe we have optional params where we can set some `days`, `hour`, and `duration` fields?